applications/vanilla/controllers/api/CategoriesApiController.php

```diff
@@ -353,6 +343,70 @@ public function post(array $body) {
     }
 
     /**
+     * Prepare data for output.
+     *
+     * @param array $row
+     */
+    public function prepareRow(array &$row) {
+        if ($row['ParentCategoryID'] <= 0) {
+            $row['ParentCategoryID'] = null;
+        }
+        $row['Description'] = $row['Description'] ?: '';
+    }
+
+    /**
+     * Move a category under a new parent.
+     *
+     * @param int $id The ID of the category.
+     * @param array $body The request body.
+     * @throws NotFoundException if unable to find the category.
+     * @throws ClientException if the parent and category are the same.
+     * @throws ClientException if the parent category ID is invalid.
+     * @throws ClientException if the target parent category does not exist.
+     * @throws ClientException if trying to move a category under one of its own children.
+     * @return array
+     */
+    public function put_parentCategoryID($id, array $body) {
+        $this->permission(['Garden.Community.Manage', 'Garden.Settings.Manage']);
+
+        $in = $this
+            ->schema(['parentCategoryID:i' => 'New parent for the category.'], 'in')
+            ->setDescription('Move a category.');
+        $out = $this->schema(['parentCategoryID:i' => 'The current parent category ID.'], 'out');
+
+        $body = $in->validate($body);
+        $parentCategoryID = intval($body['parentCategoryID']);
+
+        if ($id == $parentCategoryID) {
+            throw new ClientException('A category cannot be the parent of itself.');
+        }
+
+        if ($parentCategoryID < 1 && $parentCategoryID !== -1) {
+            throw new ClientException('parentCategoryID must be -1 or greater than zero.');
+        }
+
+        $parentCategory = $this->category($parentCategoryID, false);
+        if (!is_array($parentCategory)) {
```

Comment:
Instead of opting out of category throwing an error (there should not be an option because may be confusing down the line) you should try {} catch () {} and throw your client exception instead of testing if parentCategory is not an array.